src/diffusers/models/model_loading_utils.py

```diff
@@ -245,6 +245,9 @@ def load_model_dict_into_meta(
             ):
                 param = param.to(torch.float32)
                 set_module_kwargs["dtype"] = torch.float32
+            # For quantizers have save weights using torch.float8_e4m3fn
+            elif hf_quantizer is not None and param.dtype == getattr(torch, "float8_e4m3fn", None):
+                pass
```

Comment:
How would the param be handled in that case?

Reply:
We don't wouldn't apply any casting and the parameter would be loaded as is into the model.